src/cpp/lib/core/FlexLayout/flexlayout.cpp

```diff
@@ -165,32 +167,30 @@ void FlexLayout::performLayout() {
   if (!this->node) {
     return;
   }
+
   QRect rect = this->cachedRect;
-  if (!rect.isValid() || rect != geometry()) {
-    bool isSizeControlled = flexutils::isFlexNodeSizeControlled(this->node);
-    YGValue prevStyleMinWidth = YGNodeStyleGetMinWidth(this->node);
-    YGValue prevStyleMinHeight = YGNodeStyleGetMinHeight(this->node);
-    if (isSizeControlled) {
-      YGNodeMarkDirtyAndPropogateToDescendants(this->node);
-      YGNodeStyleSetMinHeight(this->node, rect.height());
-      YGNodeStyleSetMinWidth(this->node, rect.width());
-    }
+  if (flexutils::isFlexNodeSizeControlled(this->node)) {
+    YGNodeMarkDirtyAndPropogateToDescendants(this->node);
+    YGNodeStyleSetHeight(this->node, rect.height());
+    YGNodeStyleSetWidth(this->node, rect.width());
+  }
 
-    calculateLayout();
+  calculateLayout();
 
-    uint count = YGNodeGetChildCount(this->node);
-    for (uint i = 0; i < count; ++i) {
-      YGNode* childNode = YGNodeGetChild(this->node, i);
-      QRect childRect = flexutils::getFlexNodeGeometry(childNode);
-      FlexNodeContext* ctx = flexutils::getFlexNodeContext(childNode);
-      QLayoutItem* childItem = ctx->layoutItem();
-      childItem->setGeometry(childRect);
-    }
-    if (isSizeControlled) {
-      restoreNodeMinStyle(prevStyleMinWidth, prevStyleMinHeight);
-    }
+  QRect calculatedRect = flexutils::getFlexNodeGeometry(this->node);
```

Comment:
Amazing work! 
This is what I have not done before

```
QRect calculatedRect = flexutils::getFlexNodeGeometry(this->node);
```
For some reason I thought rect passed into this function is the truth.